spa/src/pages/Connections/Modals/DisconnectGHEServerModal.tsx

```diff
@@ -0,0 +1,181 @@
+import { useState } from "react";
+import { AxiosError } from "axios";
+import Modal, {
+	ModalBody,
+	ModalFooter,
+	ModalHeader,
+	ModalTitle,
+} from "@atlaskit/modal-dialog";
+import Button, { LoadingButton } from "@atlaskit/button";
+import {
+	BackfillPageModalTypes,
+	GitHubEnterpriseApplication,
+} from "../../../../../src/rest-interfaces";
+import SubscriptionManager from "../../../services/subscription-manager";
+
+/**
+ * NOTE: While testing in dev mode, please disable the React.StrictMode first,
+ * otherwise this modal won't show up.
+ */
+
+const DisconnectGHEServerAppModal =  ({
+	gheServer,
+	setIsModalOpened,
+	setSelectedModal,
+}: {
+	gheServer: GitHubEnterpriseApplication;
+	setIsModalOpened: (x: boolean) => void;
+	setSelectedModal: (selectedModal: BackfillPageModalTypes) => void;
+}) => {
+	const [isLoading, setIsLoading] = useState<boolean>(false);
+
+	const disconnect = async () => {
+		setIsLoading(true);
+		const response: boolean | AxiosError =
+			await SubscriptionManager.deleteGHEApp(gheServer.uuid);
+		if (response instanceof AxiosError) {
+			// TODO: Handle the error once we have the designs
+			console.error("Error", response);
+		} else {
+			setSelectedModal("DELETE_GHE_APP");
+		}
+	};
```

Comment:
Are we refetching the list of server and apps after successfully deleting the GHE server app?